example/use_cases/employer_university_flow/employer_university_flow/util.go

```diff
@@ -0,0 +1,315 @@
+package employer_university_flow
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/TBD54566975/ssi-sdk/credential"
+	"github.com/TBD54566975/ssi-sdk/credential/exchange"
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/cryptosuite"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/TBD54566975/ssi-sdk/util"
+	"github.com/sirupsen/logrus"
+)
+
+var cw = CustomStepWriter{}
+
+var (
+	UnsupportedDIDErorr = errors.New("unsupported Method for DID")
+)
+
+// Resolves a DID
+// Right the current implementation ssk-sdk does
+// not have a universal resolver.
+// https://github.com/decentralized-identity/universal-resolver
+// is a case where a universal resolver is implemented,
+// but the resolution would need to be hooked with the sdk.
+//  TODO (andor): Should exist a universal resolution method somewhere
+// in the actual SDK
+func resolveDID(didStr string) (*did.DIDDocument, error) {
+	split := strings.Split(string(didStr), ":")
+	if len(split) < 2 {
+		return nil, errors.New("invalid DID. Does not split correctly")
+	}
+	var method = split[1]
+	switch method {
+	case did.DIDKeyPrefix:
+		return did.DIDKey(didStr).Expand()
+	case did.DIDWebPrefix:
+		return did.DIDWeb(didStr).Resolve()
+	case did.PeerMethodPrefix:
+		did, _, _, err := did.DIDPeer(didStr).Resolve()
+		return did, err
+	default:
+		return nil, fmt.Errorf("%v. Got %v method", UnsupportedDIDErorr, method)
+	}
+}
+
+// Color coding to make it easier to read terminal
+const (
+	NoteColor   = "\033[1;34m%s\033[0m"
+	ActionColor = "\033[1;36m%s\033[0m"
+	StepColor   = "\033[1;33m%s\033[0m"
+	ErrorColor  = "\033[1;31m%s\033[0m"
+	DebugColor  = "\033[0;36m%s\033[0m"
+	OKColor     = "\033[0;32m%s\033[0m"
+)
+
+type Entity struct {
+	wallet *SimpleWallet
+	Name   string
+}
+
+func (e *Entity) GetWallet() *SimpleWallet {
+	return e.wallet
+}
+func NewEntity(name string, keyType string) (*Entity, error) {
+	e := Entity{
+		wallet: NewSimpleWallet(),
+		Name:   name,
+	}
+	err := e.wallet.Init(keyType)
+	if err != nil {
+		return nil, err
+	}
+	return &e, nil
+}
+
+func initSampleBody() (error, did.DID) {
+	kt := crypto.Ed25519
+	pubKey, _, err := crypto.GenerateKeyByKeyType(kt)
+	if err != nil {
+		return err, nil
+	}
+	did, err := did.PeerMethod0{}.Generate(kt, pubKey)
+	if err != nil {
+		return err, nil
+	}
+	return nil, did
+}
+
+func InitSampleDID() (error, did.DID) {
+	return initSampleBody()
+}
+
+// Custom step writer pre-formats
+// to stdout logs via steps, actions, and notes
+type CustomStepWriter struct {
+	step int
+}
+
+func (i *CustomStepWriter) Write(s string) {
```

Comment:
nit: usually make the receiver name somewhat-meaningful like (csw *CustomStepWriter)

Reply:
sure thing.